python/vyos/ifconfig/ethernet.py

```diff
@@ -417,12 +418,38 @@ def set_ring_buffer(self, rx_tx, size):
             print(f'could not set "{rx_tx}" ring-buffer for {ifname}')
         return output
 
+    def set_switchdev(self, enable):
+        ifname = self.config['ifname']
+        if not bool(re.match(r"^eth[0-9]+$", ifname)):
+            print(f'switchdev mode is only possible on physical interfaces!')
+            return
+
+        addr, code = self._popen(f'ethtool -i {ifname} | grep bus-info | awk \'{{print $2}}\'')
+        if code != 0:
+            print(f'could not resolve PCIe address of {ifname}')
+            return
+
+        enabled = False
+        state, code = self._popen(f'devlink dev eswitch show pci/0000:01:00.1  | awk \'{{print $3}}\'')
```

Comment:
Why PCI is hardcoded?  Is it expected?

Reply:
this shouldn't have been in the commit, will fix that